spec/batch_loader_spec.rb

```diff
@@ -292,4 +292,22 @@
       expect { result.to_s }.to raise_error("Oops")
     end
   end
+
+  describe '#cache' do
+    it 'caches the value in the current thread' do
+      @executed_times = 0
+      def run_test(key)
+        BatchLoader.for(key).cache do
```

Comment:
hey, thanks a lot for contributing!

I appreciate that you use the gem. I wonder if you could achieve the same without the gem? :) for example:

```rb
def run_test(key)
  Thread.current[key] ||= begin
    @executed_times += 1
    :bar
  end
end
```

it is a bit low-level implementation but I'm sure you could define some helpers in your project if you would like.